### PR TITLE
feat: implement schema store and unit test

### PR DESCRIPTION
**Pull request by poomthiti**

### CODE REVIEW

File: src/lib/stores/schema.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
+import { makeAutoObservable } from "mobx";
+import { makePersistable } from "mobx-persist-store";
+
+import type { Dict, Option } from "lib/types";
+
```
Comment (evilpeach) [inline src/lib/stores/schema.ts:5]: I would prefer to utilize the `RJSFSchema` type from `@rjsf/utils` instead.

File: src/lib/stores/schema.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+type JsonSchema = Dict<string, unknown>;
+
+// InternalMsg interface assists with accessing certain known properties without getting type errors when writing getter functions
+interface InternalSchema {
```
Comment (evilpeach) [inline src/lib/stores/schema.ts:9]: No need to declare it here.

File: src/lib/stores/schema.ts [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+  getSchemaByCodeHash(codeHash: string): Option<JsonSchema> {
+    return this.jsonSchemas[codeHash];
+  }
+
+  getSchemaProperty(codeHash: string, property: SchemaProperties) {
+    return this.jsonSchemas[codeHash]?.[property];
+  }
+
+  getQuerySchemaFormArray(codeHash: string) {
```
Comment (evilpeach) [inline src/lib/stores/schema.ts:71]: I think we can simplify this by using `getQuerySchema` and defining the output interface. This approach will make the code more clear.

File: src/lib/stores/schema.ts [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        },
+        {
+          ...response,
+          readOnly: true,
+        },
+      ];
+    });
+  }
+
+  getExecuteSchemaFormArray(codeHash: string) {
```
Comment (evilpeach) [inline src/lib/stores/schema.ts:99]: ditto

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by evilpeach — APPROVED** [review]

**Review by songwongtp — APPROVED** [review]

**Pull request merged**